### ISSUE TITLE
All helpers are included in helper object for helper specs, despite clear_helpers

### ISSUE THREAD

**Issue by ewollesen** [rspec/rspec-rails#340]
It seems that the "helper" object inside helper specs has all helper modules included into it, despite having clear_helpers in ApplicationController.

I setup a small test app using rails 3.0.5, rspec-rails 2.5.  In it, I edited my ApplicationController to call "clear_helpers."

I then created two helpers:

```
app/helpers/bar_helper.rb:

module BarHelper
  def bar
    "I am the bar helper"
  end
end
```

app/helpers/foo_helper.rb:

```
module FooHelper
  def bar
    "I am the foo helper"
  end
end
```

Then I setup the following spec in spec/helpers/bar_helper_spec.rb:

```
require "spec_helper"

describe BarHelper do
  it "should be the bar helper" do
    helper.bar.should == "I am the bar helper"
  end
end
```

When run, it fails with:

```
Failures:

  1) BarHelper should be the bar helper
     Failure/Error: helper.bar.should == "I am the bar helper"
       expected: "I am the bar helper"
            got: "I am the foo helper" (using ==)
     # ./spec/helpers/bar_helper_spec.rb:5
```

It seems that FooHelper is included in the helper object.  Is there any way to "clear_helpers" for rspec's helper object?

Thanks!

**Comment by kingdonb** [comment]
I don't think that Rails still includes all helpers everywhere (at least not custom defined ones) but RSpec still does.  I find myself needing to include helpers to get their methods in controllers (or maybe it only includes all helpers inside of views?  Helpers are supposed to be helping views, right...)

There's still a good reason to do it this way even if Rails changed its own behavior, I tried changing my spec to use only the helper I was testing as described above, and I wound up needing to do this too, because I was actually using a bunch of methods from other helpers without realizing it:

```ruby
let(:helper_object) { Object.new.
  extend(PersonnelActionHelper).
  extend(ActionView::Helpers::AssetTagHelper).
  extend(ActionView::Helpers::FormTagHelper).
  extend(ActionView::Helpers::TagHelper)
}
```

Maybe it would make sense to match the behavior of Rails?  Maybe it already does... the more you know!

**Comment by JonRowe** [comment]
@kingdonb if you'd like to take a stab at a PR I'd be happy to help review it

**Comment by kingdonb** [comment]
I did not know about this behavior of rails until I came to this issue report.  It seems I'm just wrong about how it works.

http://api.rubyonrails.org/v4.2.7.1/classes/ActionController/Helpers.html <- I am on an old version of Rails, but it seems to say the behavior described here is correct, all helpers are included in all controllers by default, unless you change `config.include_all_helpers` to return to the old behavior of matching controllers to their helpers by name.

This seems to be the same behavior described in the latest Rails version, I don't know how far back this change was made, I really just assumed that it was doing the match by name.  But then I don't know how I'm getting AssetTagHelper, FormTagHelper, and friends, which are all obviously included because they work just fine inside of my controller's views without being included manually.

If anyone is writing a project with several helpers with different methods that have the same name but different host classes, they're in for a bad time and had better know how this works.  RSpec seems to be doing the same thing that Rails has documented.  No change needed then, IMHO.
